extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerConfig.java

```diff
@@ -19,8 +20,52 @@
     /** Build / static runtime config for span exporters */
     public SpanExporterConfig exporter;
 
+    /**
+     * Tracing ID generator.
+     * <p>
+     * Valid value is {@code xray}.
+     * <p>
+     * If not set defaults to the default OpenTelemetry ID Generator.
+     */
+    @ConfigItem
+    public Optional<IdGenerator> idGenerator;
+
+    /**
+     * A comma separated list of name=value resource attributes that
+     * represents the entity producing telemetry
+     * (eg. {@code service.name=authservice}).
+     */
+    @ConfigItem
+    public Optional<List<String>> resourceAttributes;
+
+    /**
+     * Comma separated list of resources that represents the entity that is
+     * producing telemetry.
+     * <p>
+     * Valid values are {@code beanstalk, ec2, ecs, eks, host, lambda, os,
+     * process, process_runtime}.
+     */
+    @ConfigItem
```

Comment:
Similar to id generator, if we take the config approach, we're limited to what we already provide as config.

We could have the recorder look for `Resource` CDI beans and then merge the contents right?

Reply:
We can also use service loader for it later

Reply:
I'm leaning towards not defining `Resource` as a config item, but looking for CDI Bean `Resource` instances and/or service loader lookup instead.

Reply:
One thing that crossed my mind is creating Quarkus own service provider interfaces and quakiverse extensions can implement it... the OTEL SPI is waiting a `ConfigProperties` to be able to create the resources.

Reply:
The OTeL autoconfigure SPI does, but we can define that Quarkus extensions, or applications, produce one, or many, `Resource` CDI bean(s) and we look for those, correct?

Reply:
I changed to the SPI =] Let's move the convo to the there =]